webapp/src/components/assetEditor/assetSidebar.tsx

```diff
@@ -103,16 +103,19 @@ class AssetSidebarImpl extends React.Component<AssetSidebarProps, AssetSidebarSt
     }
 
     protected duplicateAssetHandler = () => {
-        pxt.tickEvent("assets.duplicate", { type: this.props.asset.type.toString(), gallery: this.props.isGalleryAsset.toString() });
+        const { isGalleryAsset } = this.props;
+        pxt.tickEvent("assets.duplicate", { type: this.props.asset.type.toString(), gallery: isGalleryAsset.toString() });
 
         const asset = this.props.asset;
-        if (!asset.meta?.displayName) asset.meta = { ...asset.meta, displayName: getDisplayNameForAsset(asset, this.props.isGalleryAsset) }
+        const displayName = asset.meta.displayName
```

Comment:
Just to confirm, asset.meta can't be null here, right? it was doing a null propagation there before but if it was null it'd have crashed on the `...asset.meta` anyway

Reply:
ah there was a null check in the if statement before! i'll add it back here, thanks!